src/Microsoft.PowerShell.Commands.Utility/commands/utility/Write-Object.cs

```diff
@@ -55,10 +55,8 @@ protected override void ProcessRecord()
             {
                 enumerate = false;
             }
-            foreach (PSObject inputObject in _inputObjects) // compensate for ValueFromRemainingArguments
-            {
-                WriteObject(inputObject, enumerate);
-            }
+
+            WriteObject(_inputObjects, enumerate);
```

Comment:
>Write-Output a,b,c
This is a symptom of the flawed binding logic I changed (assuming the first call to BindParameter would fail when the a,b,c syntax was used)

in this case, the first call to `BindParameter` is successful, because the parameter type of `InputObject` is `PSObject[]`, and type conversion from `list<object[]>` to it would be successful.

```
write-output -NoEnumerate a b c | % { $_.GetType().FullName; $_.Count; $_ -join '/' }
System.String
1
a
System.String
1
b
System.String
1
c

write-output -NoEnumerate a,b,c | % { $_.GetType().FullName; $_.Count; $_ -join '/' }
System.Object[]
3
a/b/c
```
IMHO, this behavior of `Write-Output` is pretty intuitive -- when `-NoEnumerate` is specified, give me back what ever items I gave you. In case of `a b c`, I gave you 3 items, so 3 items should be returned by `Write-Output`; in case of `a,b,c`, I gave you 1 item, which is an array, so the same item should be returned back.

However, it gets confusing when `-InputObject` is explicitly specified:
```
PS D:\> Write-Output -InputObject a,b,c -NoEnumerate | % { $_.GetType().FullName; $_.Count; $_ -join '/' }
System.String
1
a
System.String
1
b
System.String
1
c
PS D:\> Write-Output a,b,c -NoEnumerate | % { $_.GetType().FullName; $_.Count; $_ -join '/' }
System.Object[]
3
a/b/c
```
This is the part I don't like about `ValueFromRemainingArgument`, and changing it would definitely break something.